### DISCUSSION TITLE
Table cell's content is not centered despite `|:---:|` forced alignment

### DISCUSSION THREAD

**Discussion by kdeldycke** [pradyunsg/furo#279] [Questions] [answered]
### What's happening?

Here is a table in MyST:
```markdown
| Column | Left-aligned | Centered | Right-aligned |
|--------|:-------------|:--------:|--------------:|
| 1      | 2            | 3        | 4             |
| A      | B            | C        | D             |
```

I expect the content of the first and second column to be left-aligned, the third column to be centered and the last one to be right aligned.

Unfortunately all cells ends up left-aligned.

Here is the HTML produced for a row:
```html
<tr class="row-odd">
    <td>
        <p>A</p>
    </td>
    <td class="text-align:left">
        <p>B</p>
    </td>
    <td class="text-align:center">
        <p>C</p>
    </td>
    <td class="text-align:right">
        <p>D</p>
    </td>
</tr>
```

See how the alignment gets hard-coded in the form of a class attribute? This seems to be fixed if I manually edit each `<td>` tag and replace `class=` by `style=`.

### Reproducer

1. Go to the auto-generated Sphinx documentation at https://kdeldycke.github.io/meta-package-manager/#supported-package-managers and see how all tickmarks `✓` in the table are left-aligned.
2. Compare the rendered table above with the original source at https://github.com/kdeldycke/meta-package-manager/blob/develop/readme.md?plain=1#L48 in which all columns containing tickmarks `✓` are constrained to be centered.
3. Compare how GitHub render that same table, respecting the alignment: https://github.com/kdeldycke/meta-package-manager#supported-package-managers

### Expectation

Here I expect the contant of each column to respect the alignment constraint defined in Markdown between the header of the table and its rows.

### Code of Conduct

- [X] I agree to follow the [Code of Conduct](https://github.com/pradyunsg/furo/blob/main/CODE_OF_CONDUCT.md).

**Comment by pradyunsg** [comment]
This sounds like a bug to file against [myst-parser](https://github.com/executablebooks/MyST-Parser).

/cc @choldgraf

**Reply by choldgraf** [reply]
It's sorta a bug, but I think Furo will need to work around it either way. Here's the ticket tracking it:

https://github.com/executablebooks/MyST-Parser/issues/412

The challenge it seems is that docutils doesn't let you inject `style` into elements, only classes. So until that is resolved, themes will need to add support for the classes that the parser injects.

The enhancement would be to use classes without colons in them, but I think either way Furo will need to add rules for them.

**Reply by pradyunsg** [reply]
Can the parser instead use `align-center` / `align-left` / `align-right` class then, like all the other content that can be aligned (eg: images, figures)? Or use some sort of `text-center` / `text-left` / `text-right` class?

I'm happy to amend the Furo stylesheet to properly handle the table elements with that (because the `display: block` can still mess up table content in some cases, which needs fixing).

**Reply by choldgraf** [reply]
yeah I think that's the actionable thing in the issue I linked - choose a better class name than the current one 👍 . I'll bring your suggestion over there

**Comment by pradyunsg** [comment, marked as answer]
Worked around in https://github.com/pradyunsg/furo/commit/08ba46d070bfb9e0e013586822e073742dbb372c.

I'll cut a release with this, but only with a grumpy face. >.<

**Reply by choldgraf** [reply]
Why the negging? 😞 we are doing the best we can with the resources we have

**Reply by pradyunsg** [reply]
Ooops. This wasn't intended to be nagging the folks who are doing the work!

It is, rather, that I'm getting more amd more annoyed that foundational OSS stuff is under-resourced, and that's frustration leaking through here -- seemingly directed at the wrong folks!

**Reply by choldgraf** [reply]
You and me both haha, i think i just reacted strongly to your use of "bad" in the commit message. It's not fun to hear that a project you work on is "bad" first thing on a Saturday morning 🙂

**Reply by pradyunsg** [reply]
Apologies for that. 🙈

**Comment by kdeldycke** [comment]
I'm sorry for the noise. I should have checked `MyST-Parser` bug tracker first. 🙇

Feel free to ignore my bug report and close it if it doesn't make sense. I felt like reporting it for completeness but this is just nitpicking. It is a small inconvenience I can perfectly live without! 😅

BTW, thanks a lot @pradyunsg for Furo. I stumble upon it this week and it was so clean and beautiful that it was the best excuse I found to finally switch to Markdown-based documentation in Sphinx. 😍

**Comment by kdeldycke** [comment]
FYI, I confirm the latest released version of Furo fixed the issue. See these perfectly aligned columns:

<img width="810" alt="Screenshot 2021-10-12 at 12 40 49" src="https://user-images.githubusercontent.com/159718/136922861-d7e40ae6-38e2-4143-8813-9d31f4ea28d7.png">

Source: https://kdeldycke.github.io/meta-package-manager/#supported-package-managers
